python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -543,6 +543,80 @@ def upload_messages_chunk(self, payload: IO, index: int) -> None:
 # ########################
 
 
+class IContext(ABC):
+    """Base class for asset context implemented by AssetExecutionContext and ExtContext."""
+
+    @property
+    @abstractmethod
+    def is_asset_step(self) -> bool:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_key(self) -> str:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_keys(self) -> Sequence[str]:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def provenance(self) -> Optional[ExtDataProvenance]:
```

Comment:
For all methods that have a return type `Ext<something>` (ie `ExtDataProvenance`) should the return type be left empty and both implementing contexts just supply their own return type? That's a bit unsatisfying, but I'm not sure what another option is